### PR TITLE
Fix default user agent string

### PR DESCRIPTION
**Pull request by tinovyatkin**
Fixes default user agent string by reading package name, version, and homepage from `package.json` instead of using hard-coded strings.

### CODE REVIEW

File: src/request.js [hunk 1]
```diff
@@ -234,7 +238,7 @@ export function getNodeRequestOptions(request) {
 
 	// HTTP-network-or-cache fetch step 2.11
 	if (!headers.has('User-Agent')) {
-		headers.set('User-Agent', 'node-fetch/1.0 (+https://github.com/bitinn/node-fetch)');
+		headers.set('User-Agent', `${packageName}/${version} [Node ${process.version}] (+${homepage})`);
```
Comment (NotMoni) [inline src/request.js:241]: since `${version}`  is just a number. I think you should do `v${version}`

**Comment by jimmywarting** [comment]
for ref: #503 #522 #527

**Comment by jimmywarting** [comment]
https://github.com/WICG/ua-client-hints

**Comment by tinovyatkin** [comment]
> https://github.com/WICG/ua-client-hints

Do you mean to put Node version into `()`, not `[]`?

**Comment by jimmywarting** [comment]
I mean, we have had this discussion before.

> User agents identify themselves to servers as part of each HTTP request via the User-Agent header. This header's value has grown in both length and complexity over the years; a complicated dance between server-side sniffing to provide the right experience for the right devices on the one hand, and client-side spoofing in order to bypass incorrect or inconvenient sniffing on the other

UA is complex, long and sometimes unnecessary
It seems like they want to take action in reducing User-Agent Granularity, So I'm fine with keeping it at v1 and expose as little as possible too.

**Comment by tinovyatkin** [comment]
I believe that at least wrong link must be fixed. And version should be either correct or removed completely, makes no sense to _spoof_ it.
If some consumer of the library is worried about information leak then there is a documented way to prevent it.

File: src/request.js (hunk 1 above)
Comment (tinovyatkin) [reply src/request.js:230]: 'v' prefix is traditionally omitted in userAgent strings.

**Comment by LinusU** [comment]
I personally would prefer keeping the hard coded string, or even removing version information. Is there any upside to sending this information out with every request?

Also, what was the rationale for adding the Node.js version to the string?

Chrome is also moving towards hard-coding the value, and never changing it: https://groups.google.com/a/chromium.org/forum/#!msg/blink-dev/-2JIRNMWJ7s/yHe4tQNLCgAJ

Maybe I'm missing some upside to this though?

**Comment by jimmywarting** [comment]
what do you think about just using `user-agent: node-fetch` or `user-agent: node`? (trying to imagine how many bytes could be saved worldwide by 14M weekly node-fetch downloads if we shaved of 50 bytes from our current user-agent string)

> Is there any upside to sending this information out with every request?

read this: https://github.com/node-fetch/node-fetch/issues/226

**Comment by bitinn** [comment]
My vote is on a simple string.

My original design was to emulate how google crawler bot would do it:

https://support.google.com/webmasters/answer/1061943?hl=en

I am happy with `node-fetch/3.0` alone, or just `node-fetch`.

**Comment by Richienb** [comment]
As per consensus, it seems that we should just leave the UA alone as a static string.

**Comment by xxczaki** [comment]
Yes, let's change it to `node-fetch`.

**Comment by caub** [comment]
Please, could we change back the user-agent to something less long by default? (including the link to github is a good idea, but that's wasting nabwidth for most stuff and people that don't override this default)

`node-fetch` or `node-fetch/2.x` or similar

**Comment by jimmywarting** [comment]
the current user agent is just `node-fetch`...

**Pull request merged**